src/containers/survey/HourlyAppUsageSurvey.js

```diff
@@ -0,0 +1,241 @@
+// @flow
+import { useEffect, useReducer } from 'react';
+
+import { Map, Set, fromJS } from 'immutable';
+import {
+  AppContainerWrapper,
+  AppContentWrapper,
+  Card,
+  CardSegment,
+} from 'lattice-ui-kit';
+import { ReduxUtils } from 'lattice-utils';
+import { useDispatch } from 'react-redux';
+import type { RequestState } from 'redux-reqseq';
+
+import ConfirmSurveySubmissionModal from './components/ConfirmSurveySubmissionModal';
+import HourlySurvey from './components/HourlySurvey';
+import HourlyUsageSurveyAppBar from './components/HourlyUsageSurveyAppBar';
+import InstructionsModal from './components/InstructionsModal';
+import SubmissionSuccessful from './components/SubmissionSuccessful';
+import HourlySurveyDispatch, { ACTIONS } from './components/HourlySurveyDispatch';
+import { submitSurvey } from './SurveyActions';
+
+import BasicErrorComponent from '../shared/BasicErrorComponent';
+import { PROPERTY_TYPE_FQNS } from '../../core/edm/constants/FullyQualifiedNames';
+
+const { isFailure, isSuccess, isPending } = ReduxUtils;
+
+const { USER_FQN } = PROPERTY_TYPE_FQNS;
+
+const initialState = {
+  childHourlySelections: Map().asMutable(),
+  childOnlyApps: Set().asMutable(),
+  isConfirmModalVisible: false,
+  isInstructionsModalVisible: false,
+  isSubmissionConfirmed: false,
+  otherChildHourlySelections: Map().asMutable(),
+  sharedApps: Set().asMutable(),
+  step: 0,
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case ACTIONS.ASSIGN_USER: {
+      const { childOnly, appName } = action;
+
+      const { childOnlyApps, sharedApps } = state;
+
+      const selected = childOnly ? childOnlyApps : sharedApps;
+
+      if (selected.has(appName)) {
+        selected.delete(appName);
+      }
+      else {
+        selected.add(appName);
+      }
+
+      if (childOnly) {
+        return {
+          ...state,
+          childOnlyApps: selected
```

Comment:
when returning the new state, shouldn't the properties declared with `.asMutable()` be returned with `.asImmutable()`? Otherwise it seems like they're left in their .asMutable() state which defeats the purpose of using Immutable collections.

Reply:
No, they need to returned as mutable. Changing to immutable will introduce a bug. The first mutation will work fine, but subsequent mutations won't be possible.